stdlib/uuid.ksy

```diff
@@ -0,0 +1,49 @@
+meta:
+  id: uuid
+  endian: le
+  license: Unlicense
+doc: "UUID according to https://www.ietf.org/rfc/rfc4122.txt"
+seq:
+  - id: time_low
+    type: u4be
+    doc: The low field of the timestamp
+  - id: time_mid
+    type: u2be
+    doc: The middle field of thetimestamp
+  - id: time_hi_and_version
+    type: u2be
+    doc: The high field of the timestamp multiplexed with the version number
+  - id: clock_seq_hi_and_reserved
+    type: u1
+    doc: The high field of the clock sequence multiplexed with the variant
+  - id: clock_seq_low
+    type: u1
+    doc: The low field of the clock sequence
+  - id: node
+    type: u4be
+    doc: The spatially unique node identifier
+  - id: node1
+    type: u2be
+    doc: The spatially unique node identifier
+
+instances:
+  variant:
+    doc: >
+      The variant field determines the layout of the UUID.  That is, the interpretation of all other bits in the UUID depends on the setting of the bits in the variant field.  As such, it could more accurately be called a type field; we retain the original term for compatibility.  The variant field consists of a variable number of the most significant bits of octet 8 of the UUID.
+    pos: 0
+    type: b1
+    enum: variants
+    repeat: until
+    repeat-until: _ == 0
```

Comment:
This is wrong in so many ways and it just won't compile:

1. Variant-determing bits are in 8-th octet, i.e. it's `pos: 7`, not `pos: 0`.
2. You don't really need to read "until 0 is encountered", you need to just need to read at most 3 bits. In theory, 0 might never come, and this will crash with EOS exception.
3. If you use `type: b1`, it gives you a boolean value. It can't be compared to 0 in `repeat_until`, it should be `repeat-until: _ == false` or `repeat-until: not _`.
4. After using `repeat`, you're essentially getting an array of booleans, not an integer. Thus you can't apply `enum` on top of array of booleans.

But most of all, the fundamental flaw is that you're parsing a certain format first (using `seq`), and then start to analyze whether it was that format or not.